src/test/regress/expected/undistribute_table_cascade.out

```diff
@@ -0,0 +1,313 @@
+ t
+(1 row)
+
+ROLLBACK;
+BEGIN;
+  SELECT undistribute_table('citus_local_table_1', cascade=>true);
+ undistribute_table
+---------------------------------------------------------------------
+
+(1 row)
+
+  -- print foreign keys only in one of xact blocks not to make tests too verbose
+  SELECT conname, conrelid::regclass, confrelid::regclass
+  FROM pg_constraint
+  WHERE connamespace = (SELECT oid FROM pg_namespace WHERE nspname='undistribute_table_cascade') AND
+        conname ~ '^fkey\_\d+$'
+  ORDER BY conname;
+ conname |      conrelid       |      confrelid
+---------------------------------------------------------------------
+ fkey_1  | distributed_table_3 | distributed_table_2
+ fkey_11 | distributed_table_2 | distributed_table_2
+ fkey_12 | reference_table_1   | reference_table_1
+ fkey_2  | distributed_table_2 | distributed_table_3
+ fkey_3  | distributed_table_2 | distributed_table_1
+ fkey_4  | distributed_table_1 | reference_table_1
+ fkey_5  | reference_table_2   | reference_table_1
+ fkey_6  | citus_local_table_1 | reference_table_1
+ fkey_7  | reference_table_2   | citus_local_table_2
+ fkey_8  | distributed_table_1 | distributed_table_3
+(10 rows)
+
+  SELECT COUNT(*)=0 FROM pg_dist_partition, pg_tables
+  WHERE tablename=logicalrelid::regclass::text AND
+        schemaname='undistribute_table_cascade';
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+ROLLBACK;
+BEGIN;
+  SELECT COUNT(*) FROM distributed_table_1;
+ count
+---------------------------------------------------------------------
+     0
+(1 row)
+
+  -- show that we error out as select is executed in parallel mode
+  -- and there are reference tables in our subgraph
+  SELECT undistribute_table('reference_table_1', cascade=>true);
+ERROR:  cannot execute table udf because there was a parallel operation on a distributed table in transaction
+ROLLBACK;
+BEGIN;
+  set citus.multi_shard_modify_mode to 'sequential';
+  SELECT COUNT(*) FROM distributed_table_1;
+ count
+---------------------------------------------------------------------
+     0
+(1 row)
+
+  -- even if there are reference tables in our subgraph, show that
+  -- we don't error out as we already switched to sequential execution
+  SELECT undistribute_table('reference_table_1', cascade=>true);
+ undistribute_table
+---------------------------------------------------------------------
+
+(1 row)
+
+ROLLBACK;
+BEGIN;
+  DROP TABLE reference_table_1 CASCADE;
+  SELECT undistribute_table('distributed_table_2', cascade=>true);
+ undistribute_table
+---------------------------------------------------------------------
+
+(1 row)
+
+  -- as we spliited distributed_table_1,2 & 3 into a seperate subgraph
+  -- by dropping reference_table_1, we should not switch to sequential
+  -- execution
+  show citus.multi_shard_modify_mode;
+ citus.multi_shard_modify_mode
+---------------------------------------------------------------------
+ parallel
```

Comment:
surprisingly this test is flaky
https://app.circleci.com/pipelines/github/citusdata/citus/12786/workflows/2b783a34-3802-4727-978c-3f5763def1d3/jobs/219928

also, we didn't add any tests to `multi_foreign_key_relation_graph.sql` but there is one flaky test there too
https://app.circleci.com/pipelines/github/citusdata/citus/12786/workflows/2b783a34-3802-4727-978c-3f5763def1d3/jobs/219929